integrationtests/tests/fixtures/mod.rs

```diff
@@ -64,128 +64,214 @@ use real::{RealBitcoinTest, RealLightningTest};
+        let params = ServerConfigParams {
+            hbbft_base_port: base_port,
+            api_base_port: base_port + num_peers,
+            amount_tiers: amount_tiers.to_vec(),
+        };
+        let peers = (0..num_peers as u16).map(PeerId::from).collect::<Vec<_>>();
+
+        let max_evil = hbbft::util::max_faulty(peers.len());
+        let (server_config, client_config) =
+            ServerConfig::trusted_dealer_gen(&peers, max_evil, &params, OsRng::new().unwrap());
+
+        let disable_mocks = env::var("FM_TEST_DISABLE_MOCKS").is_ok();
+        let mut real_dir: Option<PathBuf> = None;
+        if disable_mocks {
+            let dir = env::var("FM_TEST_DIR").expect("Must have test dir defined for real tests");
+            real_dir = Some(PathBuf::from(dir));
+        }
 
-    let params = ServerConfigParams {
-        hbbft_base_port: base_port,
-        api_base_port: base_port + num_peers,
-        amount_tiers: amount_tiers.to_vec(),
-    };
-    let peers = (0..num_peers as u16).map(PeerId::from).collect::<Vec<_>>();
+        Self {
+            num_peers,
+            peers,
+            base_port,
+            server_config,
+            client_config,
+            disable_mocks,
+            real_dir,
+        }
+    }
 
-    let max_evil = hbbft::util::max_faulty(peers.len());
-    let (server_config, client_config) =
-        ServerConfig::trusted_dealer_gen(&peers, max_evil, &params, OsRng::new().unwrap());
+    // Builds (fed, user, bitcoin)
+    pub async fn build(&self) -> (FederationTest, UserTest, Box<dyn BitcoinTest>) {
+        let bitcoin: Box<dyn BitcoinTest>;
+        let bitcoin_rpc: Box<dyn Fn() -> Box<dyn BitcoindRpc>>;
+        let fed;
+        let user_db: Box<dyn Database>;
 
-    match env::var("FM_TEST_DISABLE_MOCKS") {
-        Ok(s) if s == "1" => {
+        if self.disable_mocks {
             info!("Testing with REAL Bitcoin and Lightning services");
-            let dir = env::var("FM_TEST_DIR").expect("Must have test dir defined for real tests");
-            let wallet_config = server_config.iter().last().unwrap().1.wallet.clone();
-            let bitcoin_rpc = bitcoincore_rpc::bitcoind_gen(wallet_config.clone());
-            let bitcoin = RealBitcoinTest::new(wallet_config);
-            let socket_gateway = PathBuf::from(dir.clone()).join("ln1/regtest/lightning-rpc");
-            let socket_other = PathBuf::from(dir.clone()).join("ln2/regtest/lightning-rpc");
-            let lightning =
-                RealLightningTest::new(socket_gateway.clone(), socket_other.clone()).await;
-            let lightning_rpc = Mutex::new(
-                ClnRpc::new(socket_gateway.clone())
-                    .await
-                    .expect("connect to ln_socket"),
-            );
 
+            let dir = self
+                .real_dir
+                .clone()
+                .expect("disable_mocks is true so the dir got initialized in the constructor");
+
+            //setup real bitcoin
```

Comment:
I usually don't point out to formatting but generally people put a space after `//`. I'm actually surprised `rustmft` doesn't change it automatically. :D

Reply:
ugh..I always catch myself doing that but I cant stop haha